go/worker/compute/executor/committee/node.go

```diff
@@ -508,6 +532,14 @@ func (n *Node) HandleNewBlockLocked(blk *block.Block) {
 
 		// Record time taken for successfully processing a batch.
 		batchProcessingTime.With(n.getMetricLabels()).Observe(time.Since(state.batchStartTime).Seconds())
+
+		// Check if we are a proposer and if so try to immediately schedule a new batch.
```

Comment:
This should probably be moved outside the switch, in case the current state after the switch is `StateWaitingForBatch`?

Reply:
hm, yeah i think that makes more sense indeed